packages/langchain/src/openai/types.ts

```diff
@@ -0,0 +1,64 @@
+import type { BaseChatModelParams } from '@langchain/core/language_models/chat_models';
+import { BaseLLMParams } from '@langchain/core/language_models/llms';
+import type {
+  ChatOpenAICallOptions,
+  OpenAIChatInput,
+  OpenAIEmbeddingsParams
+} from '@langchain/openai';
+import type { OpenAiChatCompletionParameters } from '@sap-ai-sdk/foundation-models';
+import type {
+  AzureOpenAiChatModel,
+  AzureOpenAiEmbeddingModel
+} from '@sap-ai-sdk/core';
+import type { ConfigurationOptions } from '@sap-ai-sdk/ai-api';
+
+/**
+ * Input for Text generation for OpenAi GPT.
+ */
+export type OpenAiChatModelInput = Omit<
```

Comment:
[q] Why omit these things? Why shouldn't users be allowed to set these hyperparameters when instantiating the client?

Reply:
They still are able to pass them, we omit the false type (langchains type is outdated) and add the correct type (the one we've generated), together, they form a type where these parameters can still be set, but are now (correctly) optional.

Reply:
hmm okay, but then why are we not using our type directly e.g. in the constructor?

Reply:
Because the langchain type also includes properties which are not part of the API

Reply:
For example? Are those things we actually need? Would there maybe be a benefit in using a union type of our type + additional langchain fields instead?

Reply:
Understood, but the things that don't overlap are either:
* missing from our OpenAI client (e.g. logprobs)
   * ->  and we should add them there, right?
*  Or they are missing on purpose and we don't want them (e.g. __includeRawResponse)..

Reply:
Yeah that's a good point 🤔 , I'll adjust it!